src/bootstrap/bootstrap.py

```diff
@@ -1254,6 +1270,7 @@ def bootstrap(help_triggered):
 def main():
     """Entry point for the bootstrap process"""
     start_time = time()
+    lock = acquire_lock()
```

Comment:
I think this is slightly earlier than necessary; it would be nice to print the help message before spending time acquiring the lock. I think a good spot is just before `build.update_submodules()` in `bootstrap()`.

Reply:
Moved!~